Sources/SuperValidator/Validators/Phone.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  File.swift
+//  
+//
+//  Created by christopher.mienarto on 23/03/22.
+//
+
+import Foundation
+
+public enum PhoneFormat {
+    // International Phone
+    case international
+    // North American Phone
+    case nanp
+    // European People's Party
+    case epp
+}
+
+extension SuperValidator.Option {
+    /// Phone
+    public struct Phone {
+        public let phoneFormatType: PhoneFormat
+        public let specifiedCountryCode: [String]
+
+        public init(
+            phoneFormatType: PhoneFormat = .international,
+            specifiedCountryCode: [String] = []
+        ) {
+            self.phoneFormatType = phoneFormatType
+            self.specifiedCountryCode = specifiedCountryCode
+        }
+    }
+}
+
+// MARK: - Error
+extension SuperValidator {
+    public enum PhoneError: Error, LocalizedError {
+        case invalidCountryCode
+        case invalidPhone
+        
+        public var errorDescription: String? {
+            switch self {
+            case .invalidCountryCode, .invalidPhone:
+                return nil
+            }
+        }
+    }
+}
+
+
+extension SuperValidator {
+    internal func phoneValidator(_ string: String, options: Option.Phone = .init()) -> Result<Void, PhoneError> {
+        switch options.phoneFormatType {
+        case .international:
+            guard string.matches(Regex.internationalPhoneFormat) else {
+                return .failure(PhoneError.invalidPhone)
+            }
+            
+            if options.specifiedCountryCode.isNotEmpty {
+                var phoneCode = ""
```

Comment:
maybe this one is `countryCode` more make sense than `phoneCode`

Reply:
done